workspaces/arborist/lib/query-selector-all.js

```diff
@@ -719,7 +719,10 @@ const hasAscendant = (node, compareNodes, seen = new Set()) => {
   }
 
   if (node.isTop && node.resolveParent) {
-    return hasAscendant(node.resolveParent, compareNodes)
+    if(hasAscendant(node.resolveParent, compareNodes)) {
+      /* istanbul ignore next */
+      return true
+    }
   }
```

Comment:
I suspect that this entire check can be removed (I couldn't craft a test case that would return true here) but, at the very least, we need to continue processing if `hasAscendant` returns false.